### PR TITLE
1791 initial value and continuation lines

### LINKED ISSUE #457: Continuation line
Fix the test
`TypeCobol.Test\NewTestsToFix\Parser\Samples\MidPriority\MultiLines.rdz.cbl`
and move it to Cobol85 standard test folder.

### LINKED ISSUE #1671: Bug in DataDefinition with continuation lines
**Describe the bug**
Some strange parsing errors may occur when using continuation lines in PICTURE clause.
Here are 2 cases :

**To Reproduce**
(Type)Cobol code that cause the bug : (if any)
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. DVZZMFT0.
      
       DATA DIVISION.
       WORKING-STORAGE SECTION.
      
       01 STR.
          05 data1                                              PIC    X
      -    X.
      
       PROCEDURE DIVISION.
           DISPLAY STR
           GOBACK
           .
      
       END PROGRAM DVZZMFT0.
```

Here, the PIC clause starts at column 65 and we get those error messages :
- ID:27 Message:Syntax error : A group item cannot be empty. Severity:2 Line:7
- ID:27 Message:Syntax error : extraneous input 'DISPLAYSTR' expecting {ProgramIdentification, ProgramEnd, ClassIdentification, ClassEnd, FactoryEnd, ObjectIdentification, ObjectEnd, MethodEnd, ProcedureDivisionHeader, WorkingStorageSectionHeader, LocalStorageSectionHeader, LinkageSectionHeader, FileDescriptionEntry, DataDescriptionEntry, DataRedefinesEntry, DataRenamesEntry, DataConditionEntry, ExecStatement, FunctionDeclarationEnd, GlobalStorageSectionHeader} Severity:2 Line:12
- ID:27 Message:Syntax error : extraneous input 'PROCEDURE' expecting {separator, statement starting keyword, keyword} Severity:2 Line:11

**Second example**
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. DVZZMFT0.
      
       DATA DIVISION.
       WORKING-STORAGE SECTION.
      
       01 STR.
          05 data1                                                 PIC X
      -    X.
      
       PROCEDURE DIVISION.
           DISPLAY STR
           GOBACK
           .
      
       END PROGRAM DVZZMFT0.
```

With the PIC clause completey at the the end of the line, the error changes :
- ID:27 Message:Syntax error : extraneous input 'PROCEDURE' expecting {separator, statement starting keyword, keyword} Severity:2 Line:11

**Expected behavior**
There should be no diagnostics for these cases.

### LINKED ISSUE #1791: Wrong inital values for data using continuation lines
**Describe the bug**
The initial alphanumeric value is wrong when the data description entry is written on multiple lines using continuations. According to specifications, the trailing spaces are included in the text value whereas here our parser trims them.

**To Reproduce**
(Type)Cobol code that cause the bug : (if any)
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. TCOMFL01.
       data division.
       working-storage section.
       01 var1 PIC X(120) VALUE "start1
      -    "end1".
       01 var2 PIC X(120) VALUE "start2
      *comment
      -    "end2".
       01 var3 PIC X(120) VALUE "start3
      
      -    "end3".
       01 var4 PIC X(120) VALUE "start4
      -
      -    "end4".
       01 var5 PIC X(120) VALUE "start5
      -    "extension
      -    "end5".
       procedure division.
           display var1
           display var2
           display var3
           display var4
           display var5
           goback
           .
       end program TCOMFL01.
```

Running this program on z/OS produces the following output :
```
start1                                 end1
start2                                 end2
start3                                 end3
start4                                 end4
start5                                 extension                                                   end5
```
which means the trailing spaces of continued lines are included into the value.

**Expected behavior**
Given the same code, the initial values in TC compiler should reflect what the Cobol compiler uses. Spaces should be properly included.

### PR DESCRIPTION
**Pull request by fm-117**
Fixes #457 
Fixes #1671 
Fixes #1791 

Building upon PR #1790, this new PR addresses bugs during token attribution on continued lines and also handle trailing spaces in continued values so as to create correct initial values.

**Important note: this is a breaking change for users ! As demonstrated in GenContinueInsideTypdef codegen test, the generated code is now different and as a consequence the runtime results will also be different. Although the new generated code is correct, users must be warned about this change of behavior.**

### CODE REVIEW

**Review by mayanje — APPROVED** [review]

File: Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl [hunk 1]
```diff
@@ -22,25 +22,26 @@
       *             'zfin'.
       * 01 var1 TYPE MyType.
         01 var1.
-           02 Tit PIC X(60) value 'blbablsdfklsdfksj dfjklsz ezfrzerzer'
+           02 Tit PIC X(60) value 'blbablsdfklsdfksj dfjklsz e         z
+      -    'frzerzer'
                                .
-           02 Titr PIC X(60) value 'blbablsdfklsdfksj dfjklsz ezfrzerzer
-      -    ''
+           02 Titr PIC X(60) value 'blbablsdfklsdfksj dfjklsz e        z
```
Comment (rooksdo) [inline Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl:28]: Same test as above variable

File: Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl [hunk 2]
```diff
@@ -22,25 +22,26 @@
...
-           02 Titre3 PIC X(60) value "blbablsdfklsdfksj dfjklsz ezfrzerz
-      -    "erzfin"
+           02 Titre3 PIC X(60) value "blbablsdfklsdfksj dfjklsz e      z
+      -    "frzerzer                                          zfin"
                           .
-           02 Titre4 PIC X(60) value 'blbablsdfklsdfksj dfjklsz ezfrzerz
-      -    'erzfin'
+           02 Titre4 PIC X(60) value 'blbablsdfklsdfksj dfjklsz e      z
+      -    'frzerzer                                          zfin'
                           .
```
Comment (rooksdo) [inline Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl:45]: We could reduce test to 
```cob
        01 MyGroup.
           02 Titre1 PIC X(60)  value 'blbablsdfklsdfksj dfjklsz e
      -             'zfrzerzer'.
           02 Titre2 PIC X(60)  value "blbablsdfklsdfksj dfjklsz e
      -             "zfrzerzer".
           02 Titre3 PIC X(60)  value "blbablsdfklsdfksj dfjklsz e
      -             "zfrzerzer
      -             "zfin".
           02 Titre4 PIC X(60)  value 'blbablsdfklsdfksj dfjklsz e
      -             'zfrzerzer
      -             'zfin'.
```

File: TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. DVZZMFT0.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 STR1.
+          05 data1                                              PIC    X
+      -    X.
+       01 STR2.
```
Comment (rooksdo) [inline TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt:8]: I don't see why this second picture test.

File: TypeCobol.Test/Parser/Programs/Cobol85/MultiLines.cbl [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
+     05 VAR2 PIC X(9) value '
 -        '         '.
  PROCEDURE DIVISION.
-     .
\ No newline at end of file
+     DISPLAY VAR1
+     DISPLAY VAR2
+     GOBACK
+     .
+ END PROGRAM TCOMFL01.
```
Comment (rooksdo) [inline TypeCobol.Test/Parser/Programs/Cobol85/MultiLines.cbl:19]: Why not cover this in InitialValuesContinuations.rdz.cbl  ?

File: Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl (hunk 2 above)
Comment (fm-117) [reply Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl:45]: I don't see the redundancy. Each line is different in the input source code.

File: TypeCobol.Test/Parser/Programs/Cobol85/MultiLines.cbl (hunk 4 above)
Comment (fm-117) [reply TypeCobol.Test/Parser/Programs/Cobol85/MultiLines.cbl:19]: This test is an adaptation of an earlier test that was left in the ``NewTestsToFix` directory.

File: TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt (hunk 3 above)
Comment (fm-117) [reply TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt:8]: Originally, the behavior was  different on these two different lines so I used directly the source code from the issue in the test.

File: TypeCobol.Test/Parser/Programs/Cobol85/MultiLines.cbl (hunk 4 above)
Comment (smedilol) [reply TypeCobol.Test/Parser/Programs/Cobol85/MultiLines.cbl:19]: I agree with @rooksdo, it's better to factorize these 2 tests.

File: TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt (hunk 3 above)
Comment (rooksdo) [reply TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt:8]: Maybe if the behavior is now the same we can delete thes second test.

File: Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl (hunk 2 above)
Comment (rooksdo) [reply Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl:45]: But unless I'm missing something, in the original test the two first variables are the same except for their names. Am I missing a tab or another special character by copy/pasting directly from Git to my IDE?

Comment (fm-117) [reply Codegen/test/resources/output/TypeCobol/GenContinueInsideTypdef.rdz.tcbl:45]: The two lines are different, they have different number of spaces included in the value.

File: TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt (hunk 3 above)
Comment (rooksdo) [reply TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt:8]: We should check how the parser reacts with
```cob
       01 STR2.
           05 data2                                        Pic 9(4) comp
      -    -3.
```

Comment (fm-117) [reply TypeCobol.Test/Parser/Programs/Cobol85/ContinuationLine/ContinuationPictureString.rdzMix.txt]: See d922866c0f084e1fe573589d3089411bdaaa1b7b.

File: TypeCobol.Test/Parser/Programs/Cobol85/MultiLines.cbl (hunk 4 above)
Comment (fm-117) [reply TypeCobol.Test/Parser/Programs/Cobol85/MultiLines.cbl]: See d922866c0f084e1fe573589d3089411bdaaa1b7b.

**Review by smedilol — APPROVED** [review]

**Review by rooksdo — APPROVED** [review]

**Review by mayanje — APPROVED** [review]

**Pull request merged**